kafka-client/src/main/java/dev/responsive/kafka/internal/stores/CommitBuffer.java

```diff
@@ -439,6 +439,7 @@ private boolean triggerFlush() {
           now
       );
       timeTrigger = true;
+      flushTriggers.reset();
```

Comment:
Maybe this is a bit pedantic, but why not just have it "reset" inside the `getInterval` call?

Or better yet, pull all the logic in line 436 inside an evaluation method on the FlushTriggers class (eg #shouldTriggerOnInterval) and have it test the interval + reset it with the next jitter-applied interval?

Reply:
I thought about doing it this way originally, but chose the current structure to keep the logic for what happens when a flush actually happens in a single place. So FlushTriggers stays dumb and just tells you what the triggers are, and can be reset to apply a jitter. And CommitBuffer actually decides whether or not a flush actually happens and does a reset if it actually does flush. That said, it probably makes sense to move `reset` to the branch where the flush actually happens rather than just for the time interval trigger.